d_con/source/app.d

```diff
@@ -59,23 +74,36 @@ void main()
 
 		top5[] = PostsWithSharedTags(0, 0);
 
-		ubyte minTags = 0;
-		foreach (j, count; taggedPostsCount)
+		vec_u minTags = 0;
+		ulong pv = 0;
+		vec_u eqMask = cast(ubyte)-1;
+		while (pv < nVectors)
 		{
-			if (count > minTags)
+			while (pv < nVectors && ((taggedPostsVec[pv] <= minTags) is eqMask))
+				pv++;
+			if (pv < nVectors)
 			{
-				int upperBound = TopN - 2;
+				vec_u counts = taggedPostsVec[pv];
 
-				while (upperBound >= 0 && count > top5[upperBound].sharedTags)
+				foreach (l; 0 .. vSize)
 				{
-					top5[upperBound + 1] = top5[upperBound];
-					upperBound--;
-				}
+					if (counts[l] > minTags[0])
+					{
+						int upperBound = TopN - 2;
 
-				top5[upperBound + 1] = PostsWithSharedTags(j, count);
+						while (upperBound >= 0 && counts[l] > top5[upperBound].sharedTags)
+						{
+							top5[upperBound + 1] = top5[upperBound];
+							upperBound--;
+						}
 
-				minTags = top5[TopN - 1].sharedTags;
+						top5[upperBound + 1] = PostsWithSharedTags(pv * vSize + l, counts[l]);
+
+						minTags = cast(vec_u) top5[TopN - 1].sharedTags;
+					}
+				}
 			}
+			pv++;
```

Comment:
You could likely make this a `foreach(v; taggedPostsVec)` and then use `continue` instead of the nested if statements to clean up the code a lot, and get rid of some indexing.

Then you remove all the handling of `pv`, and `v` is substituted for `taggedPostsVec[pv]`